pkg/psmdb/mongos.go

```diff
@@ -320,17 +320,31 @@ func volumes(cr *api.PerconaServerMongoDB, useConfigFile bool) []corev1.Volume {
 		})
 	}
 
-	if useConfigFile {
-		volumes = append(volumes, corev1.Volume{
-			Name: "config",
-			VolumeSource: corev1.VolumeSource{
+	if configSource.IsUsable() {
+		var vc corev1.VolumeSource
+
+		switch configSource {
+		case VolumeSourceConfigMap:
+			vc = corev1.VolumeSource{
```

Comment:
can we move this logic to `VolumeSource` method?
And call just `configSource.VolumeSource(requiredName)`

Reply:
7bdae0c